src/ROOTNTupleReader.cc

```diff
@@ -0,0 +1,212 @@
+#include "podio/ROOTNTupleReader.h"
+#include "podio/CollectionBase.h"
+#include "podio/CollectionBuffers.h"
+#include "podio/CollectionIDTable.h"
+#include "podio/GenericParameters.h"
+#include "rootUtils.h"
+
+#include "TClass.h"
+#include <ROOT/RError.hxx>
+#include <memory>
+
+namespace podio {
+
+template<typename T>
+void ROOTNTupleReader::readParams(const std::string& name, unsigned entNum, GenericParameters& params) {
+  auto keyView   = m_readers[name][0]->GetView<std::vector<std::string>>(root_utils::getGPKey<T>());
+  auto valueView = m_readers[name][0]->GetView<std::vector<std::vector<T>>>(root_utils::getGPValue<T>());
+
+  auto keys = keyView(entNum);
+  auto values = valueView(entNum);
```

Comment:
Just for my understanding. These are really just views to the data, right? There is no way we can "steal" them without destroying things. I.e. we cannot simply read this things once and then move them into the `GenericParameter` maps?

Reply:
Yep. Do you mean reading them at once for all `entNum` and then moving? I don't think that would change anything `values` here is already a vector and we are writing in a `map<string, vector<T>>`. But one thing that can be done is to cast to` an `rvalue` so that `values` can be moved in the next line:
```
  for (size_t i = 0; i < keys.size(); ++i) {
    params.getMap<T>()[keys[i]] = std::move(values[i]);
  }
```

Reply:
Yeah, my thinking was that we could provide some functionality for `GenericParameters` that simply takes the keys `vector<string>&&` and the values `vector<vector<T>>&&` and then we move them into that function and inside that function we would move the individual elements into the internal map. But maybe something like:

```cpp
params.getMap<T>().emplace(std::move(keys[i]), std::move(values[i]));
```

would be good enough. In this way we would also move the keys instead of copying them. Anyhow, all of this only works if we don't accidentally steal the things from the RNTuple view. I am not sure about that, maybe we have to make copies in any case.

Reply:
I checked and the values are being moved inside the function because the vectors are left empty but if you call `GetView` again then you get the values again which means that at least one copy is being made. So I'm actually not sure the moves will do anything. But I guess there is no downside to `std::move`. I guess we are only changing when the copies are made

Reply:
Yeah, I suppose the `GetView` will have to do some reading and potentially uncompressing. Maybe it skips this if the thing it views is non-empty. In any case, since we call `GetView` only once per event, and we don't break things by `move`-ing, I agree that we keep `std::move` here and at least avoid unnecessary copies.

Reply:
Ok this is done now moving both keys and values, I also removed the intermediate variables since they are not really used more times than that one